src/debugAdapter/transaction/transactionProvider.ts

```diff
@@ -27,39 +26,36 @@ export class TransactionProvider {
   }
 
   public async getLastTransactionHashes(take: number = TRANSACTION_NUMBER_TO_SHOW): Promise<string[]> {
-    let latestBlock: any;
+    let latestBlockNumber: number;
 
     try {
-      latestBlock = await this._web3.eth.getBlockNumber();
+      latestBlockNumber = await this._web3.eth.getBlockNumber();
     } catch {
       throw new Error(Constants.informationMessage.transactionNotFound);
     }
 
-    const initialBlock = latestBlock - take > 0 ? latestBlock - take : 0;
-    const blockNumbers = _.range(initialBlock + 1, latestBlock + 1, 1);
+    const totalBlocks = latestBlockNumber - take > 0 ? take : latestBlockNumber;
+    const startingBlockNumber = latestBlockNumber - take > 0 ? latestBlockNumber - take : 0;
+    const blockNumbers = Array.from({length: totalBlocks}, (_, i) => i + 1 + startingBlockNumber).reverse();
     const batchRequest = this._web3.createBatchRequest();
 
-    _.chain(blockNumbers)
-      .reverse()
-      .compact()
-      .value()
-      .forEach((block) => {
-        batchRequest.add(this._web3.eth.getBlock, block, true);
-      });
+    blockNumbers.forEach((block) => {
+      batchRequest.add(this._web3.eth.getBlock, block, true);
+    });
```

Comment:
You can probably drop the mutating `reverse()` call if you iterate in reverse order using `for(let i=L; i>=0; i--) { batchRequest.add(..., numbers[i],...) }`

Reply:
Thanks @cds-amal!